Bhakti Budiman Novanda

bhaktitud opened the pull request:
DB_USER=postgres
DB_PASSWORD=postgres
DB_NAME=db_ecommerce_cms
DB_HOST=127.0.0.1
DB_DIALECT=postgres
PORT=3000
SECRET=SECRET

bhaktitud committed:
ECOMMERCE WITH JEST

- register feature on testing
- server db and test ready

bhaktitud committed:
bulk delete on test file added

bhaktitud committed:
USER LOGIN ROUTE

- login route test on testing
- jwt token signature and password comparison added

bhaktitud committed:
PRODUCT Model Added

- Product model added
- role column in user's table added
- still testing on regiter and login feature

bhaktitud committed:
still working on testing

bhaktitud committed:
login with wrong email validation test

bhaktitud committed:
test error case on login

bhaktitud committed:
authentication and add product route added but not tested yet

bhaktitud committed:
add product tested work without auth, testing with outh on progres

dhimashary commented on middleware/authentication.js:
```diff
@@ -0,0 +1,46 @@
+
+        const { email, id } = decoded.id
+        if (decoded) {
+            User.findOne({
```
User ini admin kan ya?

dhimashary commented on models/user.js:
```diff
@@ -0,0 +1,63 @@
+        },
+      },
+      
+      unique : {
```
next jgn pake unique di model , bikin aja constraint unique pake ini https://sequelize.org/master/class/lib/query-interface.js~QueryInterface.html#instance-method-addConstraint

dhimashary commented on controllers/UserControllers.js:
```diff
@@ -0,0 +1,83 @@
+const { signToken } = require('../helpers/jwtoken')
+class UserController {
+    static register(req, res, next) {
+        const { email, password, role } = req.body
```
Role jangan lewat req body ! bedakan aja method untuk register user dan ADMIN, jadi udah dikasih default value

dhimashary commented on controllers/UserControllers.js:
```diff
@@ -0,0 +1,83 @@
+            res.status(201).json({
+                "email" : userCreated.email,
+                "id" : userCreated.id,
+                "role" : userCreated.role
```
disini balikan ada role , tapi di testing rolenya ga di test

dhimashary commented on test/user.test.js:
```diff
@@ -0,0 +1,176 @@
+                expect(res.body).toHaveProperty('message', 'Bad Request')
+                expect(res.body).toHaveProperty('errors', expect.any(Array))
+                expect(res.body.errors).toContain('email is required')
+                expect(res.body.errors.length).toBeGreaterThan(0)
```
ini kalo length errornya udah tau bisa langsung aja toBe(1)

bhaktitud committed:
admin sedders added

bhaktitud committed:
SERVER SIDE UPDATED

- product CRUD added (under testing)
- product Route added
- user email constraint updated

bhaktitud committed:
fetch and add product feature tested using postman

bhaktitud committed:
small code refactor

bhaktitud committed:
product test on success

bhaktitud committed:
trying to deploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
redeploy

bhaktitud committed:
npm cache cleaned redeploy

dhimashary commented:
stok sama price masih bisa minus, harusnya ke handle pas testing

dhimashary commented on test/product.test.js:
```diff
@@ -0,0 +1,136 @@
+                console.log(err);
+            });
+
+        Product.create({
```
User Create dan product itu sama2 async, baiknya jangan disuruh balapan gini, product create dimasukin ke thennya user create

bhaktitud replied:
ok sudah di perbaiki mas

dhimashary commented:
error dari test produk belum ada

dhimashary commented:
api doc belum ada

bhaktitud committed:
server checkpoint

bhaktitud committed:
server checkpoint

bhaktitud committed:
server checkpoint

bhaktitud committed:
server checkpoint

bhaktitud committed:
Product and User test has been added and fixed

bhaktitud committed:
product create moved after user create promise

bhaktitud commented:
> error dari test produk belum ada

bhaktitud committed:
product find one route added

bhaktitud committed:
authorization added

bhaktitud committed:
Update README.md

bhaktitud committed:
Update README.md

bhaktitud committed:
Update README.md

bhaktitud committed:
find or create cart for user fixed

bhaktitud committed:
Merge branch 'master' of https://github.com/bhaktitud/ecommerce_server

bhaktitud committed:
password length fixed

bhaktitud committed:
password length fixed

bhaktitud committed:
product test fixed

dhimashary commented on controllers/CartItemController.js:
```diff
@@ -0,0 +1,102 @@
+    static addToCart(req, res, next) {
+        console.log('masuk add to cart')
+        console.log(req.body.product.id, 'product id')
+        console.log(Number(req.headers.cartid), 'cart id')
```
kok cartid masuk ke headers?

dhimashary commented on controllers/CartItemController.js:
```diff
@@ -0,0 +1,102 @@
+        
+        CartItem.findOne({
+            where: {
+                ProductId: req.body.product.id,
```
kalo kaya gini bisa aja dong misal mr.A pingin beli barang , tapi ketemunya cartItem mr.B karena yang lo "where" cuman productId nya doang

dhimashary commented on controllers/CartItemController.js:
```diff
@@ -0,0 +1,102 @@
+        console.log('masuk fetch cart')
+        CartItem.findAll({
+            where: {
+                CartId: Number(req.headers.cartid)
```
kok masuk headers?

dhimashary commented on controllers/CartItemController.js:
```diff
@@ -0,0 +1,102 @@
+        CartItem.destroy({
+            where: {
+                ProductId: req.params.id,
+                CartId: Number(req.headers.cartid)
```
kok masuk headers?

dhimashary commented on controllers/CartItemController.js:
```diff
@@ -0,0 +1,102 @@
+        console.log('remove all item')
+        CartItem.destroy({
+            where: {
+                CartId: Number(req.headers.cartid)
```
kok masuk headers?

dhimashary commented:
- mungkin buat dapet cart itemnya jangan di taro headers, tapi saat kamu authentication kamu sekalian cariin juga CartItem nya user itu yang mana, sehabis itu dimasukin ke req.currentUser = { id: 1, cartId: 3 } misalnya kaya gitu

dhimashary commented:
- docs yang cart dll belom ada

bhaktitud committed:
server cartid fixed
